components/datadog/dogstatsd-standalone/k8s.go

```diff
@@ -52,6 +53,19 @@ func K8sAppDefinition(e config.CommonEnvironment, kubeProvider *kubernetes.Provi
 
 	opts = append(opts, utils.PulumiDependsOn(ns))
 
+	var imagePullSecrets corev1.LocalObjectReferenceArray
+	if e.ImagePullRegistry() != "" {
+		imgPullSecret, err := agent.NewImagePullSecret(e, namespace, opts...)
+		if err != nil {
+			return nil, err
+		}
+
+		opts = append(opts, utils.PulumiDependsOn(imgPullSecret))
```

Comment:
This isn’t needed because the `appsv1.DaemonSet` object below already has a direct reference to this `corev1.Secret`.
In pulumi, values are kind of [“futures” or “promises”](https://en.wikipedia.org/wiki/Futures_and_promises).
If a resource uses, as input, the output of another resource, then, there is already an implicit dependency between the two since the full creation of the later resource is needed to get the actual value of the parameter needed to create the former one.

```suggestion
```

Reply:
Fixed :+1: